### PR TITLE
peek messages from consumer group offset

### PR DESCRIPTION
**Pull request by huangnauh**
image

### CODE REVIEW

File: cmd/kaf/group.go [hunk 1]
```diff
@@ -277,6 +287,90 @@ var groupLsCmd = &cobra.Command{
...
+
+					count := 0
+					for {
+						select {
+						case <-cmd.Context().Done():
+							return
+						case msg := <-pc.Messages():
+							handleMessage(msg, mu)
+							count++
+							if count == limit {
```
Comment (birdayz) [inline cmd/kaf/group.go:362]: wouldn't it be better to use an absolute offset for the stop-condition, e.g. `if msg.offset >= offset+flagPeekAfter` ? (hope that condition is correct, it's easy have an off-by-one error here..

File: cmd/kaf/group.go [hunk 2]
```diff
@@ -25,16 +25,26 @@ import (
...
 	groupCmd.AddCommand(groupLsCmd)
 	groupCmd.AddCommand(groupDeleteCmd)
+	groupCmd.AddCommand(groupPeekCmd)
 	groupCmd.AddCommand(createGroupCommitOffsetCmd())
 
 	groupLsCmd.Flags().BoolVar(&noHeaderFlag, "no-headers", false, "Hide table headers")
 	groupsCmd.Flags().BoolVar(&noHeaderFlag, "no-headers", false, "Hide table headers")
+	groupPeekCmd.Flags().Int32SliceVarP(&flagPeekPartitions, "partitions", "p", []int32{}, "Partitions to peek from")
+	groupPeekCmd.Flags().IntVarP(&flagPeekBefore, "before", "B", 0, "Number of messages to peek before current offset")
+	groupPeekCmd.Flags().IntVarP(&flagPeekAfter, "after", "A", 0, "Number of messages to peek after current offset")
```
Comment (birdayz) [inline cmd/kaf/group.go:47]: let me sleep a night over the naming here, `before` and `after`..

**Comment by birdayz** [comment]
i like the idea for this command, thank you!

i think we should have a flag `--topic`. defaulting to all topics is fine though, i think.

also i suggest to have the `all-partitions` flag. i introduced it to `kaf commit`, which is pretty useful IMO!
you can look here: https://github.com/birdayz/kaf/blob/af602d92b60bbdd15b2922519fd4767fe389feea/cmd/kaf/group.go#L135

**Comment by huangnauh** [comment]
> i like the idea for this command, thank you!
> 
> i think we should have a flag `--topic`. defaulting to all topics is fine though, i think.
> 
> also i suggest to have the `all-partitions` flag. i introduced it to `kaf commit`, which is pretty useful IMO!
> you can look here:
> 
> https://github.com/birdayz/kaf/blob/af602d92b60bbdd15b2922519fd4767fe389feea/cmd/kaf/group.go#L135


* add flag `--topics`  default to all topics
* flag `--partitions` default to all partition

**Comment by birdayz** [comment]
LGTM. i thought a bit about it, and probably your approach, `-p` defaulting to all partitions, is better than my `--all-partitions`. could make sense to migrate `kaf group commit` to this new approach, but i'll keep it for now as-is.

thank you!

**Pull request merged**